egs/librispeech/ASR/conformer_ctc/ali.py

```diff
@@ -240,71 +308,66 @@ def main():
+    elif params.dataset == "train-clean-100":
+        train_clean_100_cuts = librispeech.train_clean_100_cuts()
+        dl = librispeech.train_dataloaders(train_clean_100_cuts)
+    elif params.dataset == "train-clean-360":
+        train_clean_360_cuts = librispeech.train_clean_360_cuts()
+        dl = librispeech.train_dataloaders(train_clean_360_cuts)
+    elif params.dataset == "train-other-500":
+        train_other_500_cuts = librispeech.train_other_500_cuts()
+        dl = librispeech.train_dataloaders(train_other_500_cuts)
+    elif params.dataset == "dev-clean":
+        dev_clean_cuts = librispeech.dev_clean_cuts()
+        dl = librispeech.valid_dataloaders(dev_clean_cuts)
+    else:
+        assert params.dataset == "dev-other", f"{params.dataset}"
+        dev_other_cuts = librispeech.dev_other_cuts()
+        dl = librispeech.valid_dataloaders(dev_other_cuts)
 
-    train_dl = librispeech.train_dataloaders()
-    valid_dl = librispeech.valid_dataloaders()
-    test_dl = librispeech.test_dataloaders()  # a list
-
-    ali_dir = Path(params.ali_dir)
-    ali_dir.mkdir(exist_ok=True)
-
-    enabled_datasets = {
-        "test_clean": test_dl[0],
-        "test_other": test_dl[1],
-        "train-960": train_dl,
-        "valid": valid_dl,
-    }
-    # For train-960, it takes about 3 hours 40 minutes, i.e., 3.67 hours to
-    # compute the alignments if you use --max-duration=500
-    #
-    # There are 960 * 3 = 2880 hours data and it takes only
-    # 3 hours 40 minutes to get the alignment.
-    # The RTF is roughly: 3.67 / 2880 = 0.0012743
-    #
-    # At the end, you would see
-    # 2021-09-28 11:32:46,690 INFO [ali.py:188] batch 21000/?, cuts processed until now is 836270  # noqa
-    # 2021-09-28 11:33:45,084 INFO [ali.py:188] batch 21100/?, cuts processed until now is 840268  # noqa
-    for name, dl in enabled_datasets.items():
-        logging.info(f"Processing {name}")
-        if name == "train-960":
-            logging.info(
-                f"It will take about 3 hours 40 minutes for {name}, "
-                "which contains 960 * 3 = 2880 hours of data"
-            )
-        alignments = compute_alignments(
+    logging.info(f"Processing {params.dataset}")
+    with NumpyHdf5Writer(out_ali_filename) as writer:
+        cut_set = compute_alignments(
             model=model,
             dl=dl,
+            writer=writer,
             params=params,
             graph_compiler=graph_compiler,
         )
-        num_utt = len(alignments)
-        alignments = dict(alignments)
-        assert num_utt == len(alignments)
-        filename = ali_dir / f"{name}.pt"
-        save_alignments(
-            alignments=alignments,
-            subsampling_factor=params.subsampling_factor,
-            filename=filename,
-        )
-        logging.info(
-            f"For dataset {name}, its alignments are saved to {filename}"
-        )
+
+    cut_set.to_json(out_manifest_filename)
```

Comment:
```suggestion
    cut_set.to_file(out_manifest_filename)
```

will make it easier if you ever want to switch to `jsonl` which enables lazy reading support